packages/web-components/src/drawer/drawer.ts

```diff
@@ -0,0 +1,279 @@
+      if (newValue) {
+        this.show();
+      } else {
+        this.hide();
+      }
+    }
+  }
+
+  /**
+   * @public
+   * Method called when the 'modalType' attribute changes
+   */
+  public typeChanged(oldValue: DrawerType, newValue: DrawerType): void {
+    if (newValue !== oldValue) {
+      if (newValue == DrawerType.inline) {
+        this.modalType = DrawerModalType.nonModal;
+      }
+    }
+  }
+
+  /**
+   * Opens the drawer if it is currently hidden.
+   * @public
+   */
+  public show(): void {
+    if (!this.dialog.open) {
+      if (this.type === DrawerType.inline || this.modalType === DrawerModalType.nonModal) {
+        this.dialog.show();
+      } else {
+        this.dialog.showModal();
+      }
+      this.open = true;
+      this.closing = false;
+      this.triggerAnimation();
+    }
+  }
+
+  /**
+   * Closes the drawer if it is currently open.
+   * @public
+   */
+  public hide(): void {
+    if (this.dialog.open) {
+      this.closing = true;
+      Updates.enqueue(() => {
+        this.triggerAnimation();
+      });
+    }
+  }
+
+  /**
+   * Handles click events on the drawer.
+   *
+   * @param event - The click event
+   * @returns boolean - Always returns true
+   * @public
+   */
+  public clickHandler(event: Event): boolean {
+    event.preventDefault();
+    if (this.open && this.modalType !== DrawerModalType.alert && event.target === this.dialog) {
+      this.hide();
+      this.dismiss();
+    }
+    return true;
+  }
+
+  /**
+   * @public
+   * Handles keydown events on the drawer
+   * @param e - The keydown event
+   * @returns boolean | void
+   */
+  public keydownHandler = (event: KeyboardEvent): boolean | void => {
+    if (event.defaultPrevented) {
+      return;
+    }
+    switch (event.key) {
+      case keyEscape:
+        event.preventDefault();
+
+        if (this.modalType !== DrawerModalType.alert) {
+          this.hide();
+          this.dismiss();
+        }
+        break;
+      default:
+        return true;
+    }
+  };
+
+  /**
+   * A function that calls the animation end handler.
+   * @private
+   */
+  private readonly animationEndHandlerFunction = (): void => this.animationEndHandler();
+
+  /**
+   * Triggers the opening or closing animation on the drawer.
+   * @private
+   */
+  private triggerAnimation(): void {
+    this.setAttribute('data-animating', '');
+    if (this.closing) {
+      this.setAttribute('data-closing', '');
+    }
+    this.dialog.addEventListener(eventAnimationEnd, this.animationEndHandlerFunction);
+  }
+
+  /**
+   * Handles the end of the animation.
+   * @private
+   *
+   */
+  private animationEndHandler(): void {
+    this.dialog.removeEventListener(eventAnimationEnd, this.animationEndHandlerFunction);
+    this.removeAttribute('data-animating');
+    if (this.closing) {
+      this.removeAttribute('data-closing');
+      this.dialog.close();
+      this.open = false;
+      this.closing = false;
+    }
+  }
```

Comment:
I'd like to see if we can follow the pattern set forth in the open dialog PR which handles all animations within CSS.

Reply:
I ripped out all the JS and moved the animation styles to CSS